src/lib/db-listeners/db-publications-listener.ts

```diff
@@ -55,34 +55,33 @@ export class DBPublicationsListener {
         .then(doc => {
           documents.add(doc);
           onPatch(doc.stars, patch => {
-            const path = patch.path.split("/");
+            const [, index, replaceKey] = patch.path.split("/");
             if (patch.op === "add") {
               const star = patch.value;
               const { starred, key } = star;
               if (key === "") { // key will be assigned from Firebase node id
                 this.db.createUserStar(doc, starred);
               }
-            } else if (patch.op === "replace" && path[path.length - 1] === "starred") {
-              const starIndex = parseInt(path[path.length - 2], 10);
+            } else if (patch.op === "replace" && replaceKey === "starred") {
+              const starIndex = parseInt(index, 10);
               const star = doc.getUserStarAtIndex(starIndex);
               if (star) {
                 this.db.setUserStarState(doc.key, star.key, star.starred);
               }
             }
           });
           onPatch(doc.comments, patch => {
-            const path = patch.path.split("/");
-            const tileId = path[1];
+            const [, tileId, , index, replaceKey] = patch.path.split("/");
```

Comment:
FYI: there's a [trick](https://medium.com/the-andela-way/this-trick-will-help-you-de-structure-javascript-arrays-in-a-better-cleaner-way-a3bbc39ce226) which allows this to be replaced with
```
const { 1: tileId, 3: index, 4: replaceKey } = patch.path.split("/");
```
thus avoiding the need to count commas. Of course, the array indices are just as subject to change as are the commas and object destructuring of arrays is not a familiar pattern, so this may be too-clever-by-half. Consider it a point of interest rather than a recommendation.

Reply:
Ah, handy trick!